rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java

```diff
@@ -0,0 +1,534 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeMainNetConstants;
+import co.rsk.config.BridgeTestNetConstants;
+import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
+import co.rsk.peg.bitcoin.BitcoinTestUtils;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import org.bouncycastle.util.encoders.Hex;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static co.rsk.peg.PegTestUtils.createFederation;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class PegUtilsTest {
```

Comment:
Transaction with multiple outputs to the active federation? Outputs to the active and retiring federation? Outputs to the federation and to other addresses? Sent from the retiring federation to the active but not in the index (weird case but should be treated as a peg-in)

Reply:
Done. Migration txs but not in the index were there already.